SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/Storage.cs

```diff
@@ -283,6 +286,27 @@ private void InitializeDatabase()
             }
         }
 
+        private void EnsureDatabaseDirectoryExists()
+        {
+            var databaseDirectoryPath = Path.GetDirectoryName(_databasePath);
+            if (string.IsNullOrEmpty(databaseDirectoryPath))
+            {
+                return;
+            }
+            try
+            {
+                var databaseDirectory = new Directory(databaseDirectoryPath);
+                if (!databaseDirectory.Exists())
+                {
+                    databaseDirectory.Create();
+                }
+            }
+            catch (Exception e)
+            {
+                AppCenterLog.Error(AppCenterLog.LogTag, "Failed to create database directory.", e);
```

Comment:
On second thought why are we not throwing this one and handle the exception in the existing function?
If we catch it here and directory failed to be created, then storage initialization will also fail, so 2 exceptions instead of one.

Reply:
it can fails on try to access the directory but open the db file just fine. "Exist" check fixes it in this case, but anyway it does not mean that storage init will fail

Reply:
This is a very weird scenario then, can we add comments?

Reply:
Added comment